lib/fast_jsonapi/serialization_core.rb

```diff
@@ -55,6 +55,13 @@ def ids_hash_from_record_and_relationship(record, relationship)
         id_hash_from_record associated_object, polymorphic
       end
 
+      def fetch_ids(record, id_method_name, object_method_name)
+        record.public_send(id_method_name)
+
+      rescue NoMethodError
+        record.public_send(object_method_name).map(&:id)
```

Comment:
Looks like we are making an assumption that we can use `id` method to get the id of the object. if the object uses a different method to get id this would not work. Is there a way we can cover that case too?

Reply:
Thank you so much for your feedback! You are absolutely right!
I assumed that I can get id by calling `id` method but it may not be that simple..

I have some idea how to solve the problem so let me work on it!

Reply:
I added `id_from_origin` option to `has_many`, `has_one`, and `belongs_to`.
If you want to fetch id directly from the target object, all you need to do is simply set `id_from_origin: true`, and then `id_method_name` will be called upon the target objects.

I just want to see what you think about the idea so I haven't added tests or documents, but once it settles, I'll work on them! 👍 Oh, and also I will clean up the commits, too!

Reply:
Need some more time to think about the repercussions of this approach. So might take some more time to get back to you

Reply:
Sure, sure. Please let me know what you think whenever you have time! 👍